src/soft_start.cpp

```diff
@@ -6,35 +6,62 @@
 
 
 #include "kodlab_mjbots_sdk/soft_start.h"
+#include "kodlab_mjbots_sdk/log.h"
 
 namespace kodlab {
 
-void SoftStart::Constrain(std::vector<float> &values,
+void TorqueLimiter::Constrain(std::vector<float> &values,
                           float min_val,
                           float max_val) {
   for (auto &value : values) {
-    value = fmin(fmax(value, min_val), max_val);
+    value = std::min(std::max(value, min_val), max_val);
   }
 }
 
-float SoftStart::Constrain(float values, float min_val, float max_val) {
-  return fmin(fmax(values, min_val), max_val);
+float TorqueLimiter::Constrain(float values, float min_val, float max_val) {
+  return std::min(std::max(values, min_val), max_val);
 }
 
-void SoftStart::ConstrainTorques(std::vector<float> &torques,
-                                 float time_since_start_ms) {
-  if (time_since_start_ms > duration_ms_) {
-    Constrain(torques, -max_torque_, max_torque_);
-  } else {
-    float max_val = time_since_start_ms * slope_;
-    Constrain(torques, -max_val, max_val);
+void SoftStart::ConstrainTorques(std::vector<float> &torques) const {
+  if(timer_initialized_){
+    float time_since_start_ms = SoftStart::timer_.tac()/1000.0;
+    if (time_since_start_ms > duration_ms_) {
+      TorqueLimiter::Constrain(torques, -max_torque_, max_torque_);
+    } else {
+      float max_val = time_since_start_ms * slope_;
+      TorqueLimiter::Constrain(torques, -max_val, max_val);
+    }
+  }else{
+    LOG_ERROR("Soft start not initialized, constraining torque to zero");
+    TorqueLimiter::Constrain(torques, -0, 0);
   }
+}
 
+void SoftStart::ConstrainTorque(float &torque) const {
+  if(timer_initialized_) {
+    float time_since_start_ms = SoftStart::timer_.tac() / 1000.0;
+    if (time_since_start_ms > duration_ms_) {
+      TorqueLimiter::Constrain(torque, -max_torque_, max_torque_);
+    } else {
+      float max_val = time_since_start_ms * slope_;
+      torque = TorqueLimiter::Constrain(torque, -max_val, max_val);
+    }
+  } else{
+    LOG_ERROR("Soft start not initialized, constraining torque to zero");
+    torque = 0;
+  }
 }
```

Comment:
Could use a simple template and an overloading of ConstrainTorques instead of a separate ConstrainTorque to avoid rewriting this code here